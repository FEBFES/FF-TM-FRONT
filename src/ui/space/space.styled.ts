import styled from 'styled-components';
import { SizeProp, DirectionType } from './space';

export type StyledVariants<E extends string> = {
  [key in E]?: string;
};

const sizes: StyledVariants<SizeProp> = {
  '2xs': '3px',
  xs: '6px',
  s: '8px',
  sm: '12px',
  m: '16px',
  l: '24px',
  xl: '32px',
};

export const SAvatar = styled.div<{
  size: SizeProp;
  direction: DirectionType;
}>`
  margin-left: ${({ direction, size }) =>
    direction === 'row' ? sizes[size] : '0'};
  margin-right: ${({ direction, size }) =>
    direction === 'row' ? sizes[size] : '0'};

  margin-top: ${({ direction, size }) =>
    direction === 'col' ? sizes[size] : '0'};
  margin-bottom: ${({ direction, size }) =>
    direction === 'col' ? sizes[size] : '0'};

  display: ${({ direction }) =>
    direction === 'row' ? 'inline-block' : 'block'};
`;
