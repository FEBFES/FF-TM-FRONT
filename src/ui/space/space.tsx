import React from 'react';
import { SAvatar } from './space.styled';

export type SizeProp = '2xs' | 'xs' | 's' | 'sm' | 'm' | 'l' | 'xl';
export type DirectionType = 'col' | 'row';

interface SpaceProps {
  size?: SizeProp;
  direction?: DirectionType;
}

export const Space: React.FC<SpaceProps> = ({
  size = 's',
  direction = 'col',
}): JSX.Element => {
  return <SAvatar size={size} direction={direction} />;
};
