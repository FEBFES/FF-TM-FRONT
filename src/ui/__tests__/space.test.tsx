import React from 'react';
import { Space } from '../space/space';
import renderer from 'react-test-renderer';

describe('<Space />', () => {
  it('SNAPSHOTS: UI - Space (col xl) component - render', () => {
    const elem = renderer
      .create(
        <div>
          <p>123</p>
          <Space size="xl" direction="col" />
          <span>asd</span>
        </div>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Space (row s) component - render', () => {
    const elem = renderer
      .create(
        <div>
          <p>123</p>
          <Space size="s" direction="row" />
          <span>asd</span>
        </div>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });
});
