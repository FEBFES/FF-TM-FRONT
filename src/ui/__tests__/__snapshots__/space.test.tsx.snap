// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Space /> SNAPSHOTS: UI - Space (col xl) component - render 1`] = `
<div>
  <p>
    123
  </p>
  <div
    className="sc-bcPKhP cbBkoU"
    direction="col"
    size="xl"
  />
  <span>
    asd
  </span>
</div>
`;

exports[`<Space /> SNAPSHOTS: UI - Space (row s) component - render 1`] = `
<div>
  <p>
    123
  </p>
  <div
    className="sc-bcPKhP kWIBqY"
    direction="row"
    size="s"
  />
  <span>
    asd
  </span>
</div>
`;

exports[`SNAPSHOTS: UI - Space (col xl) component - render 1`] = `
<div>
  <p>
    123
  </p>
  <div
    className="sc-bcPKhP cbBkoU"
    direction="col"
    size="xl"
  />
  <span>
    asd
  </span>
</div>
`;

exports[`SNAPSHOTS: UI - Space (row s) component - render 1`] = `
<div>
  <p>
    123
  </p>
  <div
    className="sc-bcPKhP kWIBqY"
    direction="row"
    size="s"
  />
  <span>
    asd
  </span>
</div>
`;
